pkg/ebpf/net_events.go

```diff
@@ -230,40 +241,50 @@ func (t *Tracee) processNetEvents(ctx gocontext.Context) {
 					continue
 				}
 
-				if t.config.Debug {
-					evt, err := netPacketProtocolHandler(netDecoder, netEventMetadata, networkThread, "net_packet")
-					if err != nil {
-						t.handleError(err)
-						continue
-					}
-
-					// output the event
-					select {
-					case t.config.ChanEvents <- evt:
-						t.stats.NetEvCount.Increment()
-					case <-ctx.Done():
-						return
+				ifaceName := t.netInfo.ifaces[int(netCaptureData.InterfaceIndex)].Name
+				ifaceIdx, err := t.getTracedIfaceIdx(ifaceName)
+				if err == nil {
+					if t.eventsToTrace[netEventMetadata.NetEventId] && ifaceIdx >= 0 {
+						evt, err := netPacketProtocolHandler(netDecoder, netEventMetadata, networkThread, "net_packet")
+						if err != nil {
+							t.handleError(err)
+							continue
+						}
+						// output the event
+						select {
+						case t.config.ChanEvents <- evt:
+							t.stats.NetEvCount.Increment()
+						case <-ctx.Done():
+							return
+						}
 					}
 				}
 
-				// capture the packet
-				packetBytes, err := getPacketBytes(netDecoder, netCaptureData.PacketLength)
+				ifaceName = t.netInfo.ifaces[int(netCaptureData.InterfaceIndex)].Name
+				ifaceIdx, err = t.getCapturedIfaceIdx(ifaceName)
 				if err != nil {
-					t.handleError(err)
 					continue
 				}
-				if err := t.writePacket(netCaptureData, time.Unix(0, int64(netEventMetadata.TimeStamp)), packetContext, packetBytes); err != nil {
-					t.handleError(err)
-					continue
+				if ifaceIdx >= 0 {
```

Comment:
Again, it is weird to check it if there was not error (unless you expect to get -1 somehow?)